data/transactions/logic/eval.go

```diff
@@ -3615,153 +3569,128 @@ func opAppGlobalGetEx(cx *EvalContext) {
 
 	cx.stack[prev] = result
 	cx.stack[last] = isOk
+	return nil
 }
 
-func opAppLocalPut(cx *EvalContext) {
+func opAppLocalPut(cx *EvalContext) error {
 	last := len(cx.stack) - 1 // value
 	prev := last - 1          // state key
 	pprev := prev - 1         // account
 
 	sv := cx.stack[last]
 	key := string(cx.stack[prev].Bytes)
 
-	if cx.Ledger == nil {
```

Comment:
I understand this is a repetitive check... is it prereq now?

Reply:
Yes, `EvalContract` does the check when first entering stateful evaluation, so it seems pointless to keep checking it in the stateful ops.